frontend/src/my-books/MyBooks.tsx

```diff
@@ -3,236 +3,280 @@ The book project lets a user keep track of different books they would like to re
 reading, have read or did not finish.
 Copyright (C) 2020  Karan Kumar
 
+
+
+
 This program is free software: you can redistribute it and/or modify it under the terms of the
 GNU General Public License as published by the Free Software Foundation, either version 3 of the
 License, or (at your option) any later version.
 
+
+
+
 This program is distributed in the hope that it will be useful, but WITHOUT ANY
 WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
 PURPOSE.  See the GNU General Public License for more details.
 
+
+
+
 You should have received a copy of the GNU General Public License along with this program.
 If not, see <https://www.gnu.org/licenses/>.
 */
-
 import React, { Component, ReactElement } from "react";
 import { NavBar } from "../shared/navigation/NavBar";
 import Switch from "../settings/Switch";
-import Button from "@material-ui/core/Button";
-import ShelfModal from "./ShelfModal";
 import { Layout } from "../shared/components/Layout";
 import BookList from '../shared/book-display/BookList';
 import { Book } from '../shared/types/Book';
 import HttpClient from '../shared/http/HttpClient';
 import Endpoints from '../shared/api/endpoints';
-import "./MyBooks.css";
 import ShelfView from "../shared/book-display/ShelfView";
+import ShelfModal from "./ShelfModal";
+import "./MyBooks.css";
+import BookModal from "./BookModal"; // Import BookModal component
+import Button from "@material-ui/core/Button"; // Import Button component
 
 
 interface IState {
-    showShelfModal: boolean;
-    showListView: boolean;
-    bookList: Book[];
-    favoriteBooks: Book[];
-    readBooks: Book[];
-    didNotFinishBooks: Book[];
-    toReadBooks: Book[];
-    readingBooks: Book[];
-    searchVal: string;
+   showShelfModal: boolean;
+   showBookModal: boolean; // Add showBookModal state
+   showListView: boolean;
+   bookList: Book[];
+   favoriteBooks: Book[];
+   readBooks: Book[];
+   didNotFinishBooks: Book[];
+   toReadBooks: Book[];
+   readingBooks: Book[];
+   searchVal: string;
 }
```

Comment:
Hi, could you keep the theoretically unchanged lines exactly unchanged? Now many theoretically unchanged lines are recognized as changed as some unnecessary changes like extra spaces are introduced.

Reply:
Consequently, a large proportion of the review work is to identify which changes are intended changes now.